platform/engineering/request-for-comment/2024-09-23-evidence-collection-and-polling.md

```diff
@@ -0,0 +1,136 @@
+
+Single source of truth for document requirements eliminates or greatly reduces the chance of us accepting a document, and then not being able to send it somewhere. We also can (if desired) reject the file and report the reason why, live on the frontend to the end user, if it is an actionable error. 
+We will need to decide what to do in case the validation endpoint is not reachable, between taking it and hope it works later, or disallow uploads at that time since we cannot ensure it will succeed. 
+
+CarrierWave should be used to upload evidence to AWS S3, and should also include virus scanning. Example [here.](https://github.com/department-of-veterans-affairs/vets-api/blob/f9b931187f2525bf3c5187b37731286e9d1b6d16/app/uploaders/form1010cg/poa_uploader.rb) 
+
+#### Document Storage Location
+The document has to live somewhere when it is uploaded. For some amount of time this is usually the filesystem of the vets-api server. The document can be there temporarily but all efforts should be made for it to upload to S3 and reference to it be kept in the DB so it can be pulled down for use later. The filesystem is not reliable enough to be used as a store for user submitted files. It goes away on pod reboots and deploys, which could cause any evidence stored on the filesystem to be lost.  
+
+
+### Transmittal of Benefits Documents
+
+#### Authenicated/LOA3 User Document Uploads
+[Benefits Documents API](https://developer.va.gov/explore/api/benefits-documents/docs?version=current) is preferred as it directly puts the documents into the system(s) of record (VBMS/BGS).
+
+[Benefits Intake API](https://developer.va.gov/explore/api/benefits-intake/docs?version=current) goes to Central Mail, similar to if something was mailed in or faxed, it has more human and system touch-points and therefore is slower and more prone to error(s). 
+
+For some documents, this is required, such as 21-4142, since it requires manual action be done by a person. This is also appropriate for documents which fail to go the primary/Documents API path (due to error, data issues/discrepancies, downtimes, network issues, etc)
+
+
+#### Unauthenticated User Document Uploads
+As of now we are not sure the unauthenticated document upload should be any different that authenitcated. We still have a duty to ensure that the document gets where it was going, and potentially a duty to notify the uploader if we cannot.
+
+
+#### Other considerations
+All documents that cannot go their primary path (Documents API) should have a backup path/failsafe by attempting submission via the benefits intake API, unless your primary path is the benefits API or something that uses the benefits API downstream.
```

Comment:
benefits API or something that uses the benefits API

should say 

Benefits Intake API or something that uses the Benefits Intake API